src/MongoDB.EntityFrameworkCore/Storage/MongoDatabaseCreator.cs

```diff
@@ -36,16 +37,16 @@ public bool EnsureDeleted()
         => clientWrapper.DeleteDatabase();
 
     /// <inheritdoc/>
-    public Task<bool> EnsureDeletedAsync(CancellationToken cancellationToken = new())
+    public Task<bool> EnsureDeletedAsync(CancellationToken cancellationToken = default)
         => clientWrapper.DeleteDatabaseAsync(cancellationToken);
 
     /// <inheritdoc/>
     public bool EnsureCreated()
-        => clientWrapper.CreateDatabase(currentDbContext.Context.Model);
+        => clientWrapper.CreateDatabase(currentDbContext.Context.GetService<IDesignTimeModel>());
 
     /// <inheritdoc/>
-    public Task<bool> EnsureCreatedAsync(CancellationToken cancellationToken = new())
-        => clientWrapper.CreateDatabaseAsync(currentDbContext.Context.Model, cancellationToken);
+    public Task<bool> EnsureCreatedAsync(CancellationToken cancellationToken = default)
+        => clientWrapper.CreateDatabaseAsync(currentDbContext.Context.GetService<IDesignTimeModel>(), cancellationToken);
```

Comment:
Why this change?

Reply:
The design-time model contains the indexes but they are removed in the optimized run-time models.

While brekaing, the IMongoClientWrapper interface does state that this interface is for internal use and is not subject to the breaking change policy.

Reply:
Good to know!